ignite/base/mixins.py

```diff
@@ -22,3 +23,52 @@ def load_state_dict(self, state_dict: Mapping) -> None:
         opts = [k in state_dict for k in self._state_dict_one_of_opt_keys]
         if len(opts) > 0 and ((not any(opts)) or (all(opts))):
             raise ValueError("state_dict should contain only one of '{}' keys".format(self._state_dict_one_of_opt_keys))
+
+
+class Attachable:
+    """
+    This class contains an internal dictionary `attach_map` associating events with the bound methods plus args/kwargs
+    to be triggered on those events. This dictionary can be populated by the contructor or by external clients after
+    construction and before `attach` is called.
+    """
+
+    def __init__(self, attach_map={}):
+        self.attach_map = dict(attach_map)
+
+    def _iter_handlers(self):
+        for event, (method, _, _) in self.attach_map.items():
+            yield event, method
+
+    def set_handler_mapping(self, event_name: Any, handler: Callable, *args, **kwargs) -> None:
+        if handler is not None:
```

Comment:
Why do you remove event if handler is None ?

Reply:
This was a way of combining the operation of adding or removing a handler into one method. It might be better to introduce a `remove_handler_mapping` method and removal code there, or change line 45 to `elif event_name in self.attach_map` so that the method silently does nothing when `event_name` is for an event not attached yet. This could be useful if `handler` is an argument in the calling routine which is `None` by default indicating no handler for a particular event, it lets you define a do-nothing default in that case.